python/test/ibis/test_ibis_substrait.py

```diff
@@ -0,0 +1,111 @@
+from collections import OrderedDict
+
+import ibis
+import pytest
+from google.protobuf import json_format
+
+from ibis_substrait.compiler.translate import translate
+from ibis_substrait.proto.substrait import type_pb2 as stt
+from ibis_substrait.proto.substrait.algebra_pb2 import Expression, Rel
+
+NULLABILITY_NULLABLE = stt.Type.Nullability.NULLABILITY_NULLABLE
+NULLABILITY_REQUIRED = stt.Type.Nullability.NULLABILITY_REQUIRED
+
+# Test adopted from upstream ibis-substrait unit(s)
+# https://github.com/ibis-project/ibis-substrait/blob/main/ibis_substrait/tests/compiler/test_compiler.py
+
+@pytest.fixture
+def t0():
+    return ibis.table(
+        [
+            ("full_name", "string"),
+            ("age", "int64"),
+            ("ts", "timestamp('UTC')"),
+            ("delta", "interval"),
+        ]
+    )
+
+@pytest.fixture
+def t1():
+    return ibis.table(
+        [
+            ("full_name", "string"),
+            ("age", "int64"),
+            ("ts", "timestamp('UTC')"),
+            ("delta", "interval"),
+        ]
+    )
+
+
+def to_dict(message):
+    """Print Protobuf message as python dictionary object."""
+    return json_format.MessageToDict(message)
+
+
+def test_join(t0, t1, compiler):
+    """A walkthrough of a join expression in Substrait."""
+    expr = (
+        t0.left_join(t1, t0.age == t1.age)
+    )
+    result = translate(expr, compiler)
+
+    # This plan is a "volcano" style plan meant for bottoms-up execution.
+    # As a result, we top-level operation in the relation is the final projection
+    # https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto
+    #
+    # TODO(bsarden): Find out which logical plan optimizers are used.
+    assert(result.WhichOneof("rel_type") == "project")
```

Comment:
Is there a way to iterate over the relation tree without querying the type like you have here?

Reply:
I believe the expectation on the protobuf side of things is access the values dynamically using python `getattr` function.

For example:

```python
data = getattr(result, result.WhichOneof("rel_type")).value
```

Link to where this example comes from: 
https://stackoverflow.com/questions/51421392/dynamically-access-oneof-value-from-a-protobuf

Reply:
makes sense thanks!